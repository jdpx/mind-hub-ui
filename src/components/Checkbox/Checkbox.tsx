import React from 'react'

import emptyCheckbox from '../../assets/checkbox-empty.svg'
import tickedCheckbox from '../../assets/checkbox-ticked.svg'

interface Props {
    checked?: boolean
    testid?: string
}

export default function Checkbox({ checked, testid }: Props) {
    return !checked ? (
        <img
            src={emptyCheckbox}
            alt={`Empty checkbox`}
            height={50}
            data-testid={`${testid}-empty`}
        />
    ) : (
        <img
            src={tickedCheckbox}
            alt={`Ticked checkbox`}
            height={29}
            data-testid={`${testid}-checked`}
        />
    )
}
