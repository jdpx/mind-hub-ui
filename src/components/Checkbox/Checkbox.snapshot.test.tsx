import React from 'react'
import { render } from '@testing-library/react'
import Checkbox from './Checkbox'

describe('Checkbox Snapshots', () => {
    it('renders checked checkbox', () => {
        const { asFragment } = render(<Checkbox checked />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('renders unchecked checkbox', () => {
        const { asFragment } = render(<Checkbox />)
        expect(asFragment()).toMatchSnapshot()
    })
})
