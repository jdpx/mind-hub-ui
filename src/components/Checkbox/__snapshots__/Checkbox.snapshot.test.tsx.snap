// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Snapshots renders checked checkbox 1`] = `
<DocumentFragment>
  <img
    alt="Ticked checkbox"
    data-testid="undefined-checked"
    height="29"
    src="checkbox-ticked.svg"
  />
</DocumentFragment>
`;

exports[`Checkbox Snapshots renders unchecked checkbox 1`] = `
<DocumentFragment>
  <img
    alt="Empty checkbox"
    data-testid="undefined-empty"
    height="50"
    src="checkbox-empty.svg"
  />
</DocumentFragment>
`;
